Extension/src/LanguageServer/client.ts

```diff
@@ -812,6 +812,20 @@ export class DefaultClient implements Client {
                     this.semanticTokensLegend = new vscode.SemanticTokensLegend(tokenTypesLegend, tokenModifiersLegend);
 
                     if (firstClient) {
+                        // Don't log telemetry for machineId if it's a special value used by the dev host: someValue.machineid
+                        if (vscode.env.machineId !== "someValue.machineId") {
```

Comment:
I was thinking we'd do this during activation instead of during client creation.  Were you thinking it would be better to ignore people who aren't using C++ anymore?  I would prefer to have a single "activate" event with Properties:
* remoteName
* oldMachineId - optional (machineIdPersistentState.Value)
* newMachineId - optional (vscode.env.machineId)

The two last properties are only logged when the value changes.  Could you check and see whether this triggers when using Remote-WSL or one of the other Remote extensions?